pkg/controller/resourceactuation/resourceactuation.go

```diff
@@ -15,14 +15,86 @@
 package resourceactuation
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/reconciliationinterval"
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
 
+	opv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/operator/pkg/apis/core/v1beta1"
+	opk8s "github.com/GoogleCloudPlatform/k8s-config-connector/operator/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// todo acpana: revisit
+var DefaultActuationAction = ActuationAction("reconciling")
+
+type ActuationAction string
+
+func (aa ActuationAction) String() string {
+	return strings.ToLower(string(aa))
+}
+
+func defaultOrLower(s string) string {
+	if s == "" {
+		return DefaultActuationAction.String()
+	}
+
+	return strings.ToLower(s)
+}
+
+func ReconcilingByDefault() bool {
+	return DefaultActuationAction == "reconciling"
+}
+
+// ShouldActuate checks if the resource should be actuated. It errors out if it cannot fetch the
+// ConfigConnector (CC) resource or the ConfigConnectorContext (CCC) resource for the resource's namespace if running
+// in namespaced mode.
+// - If both CC & CCC specify an actuationAction in namespaced mode, we defer to the CCC's value.
+//   - If only CC specifies an actuationAction in namespaced mode, we defer to the CC's value.
+// - If both CC & CCC specify an actuationAction in cluster mode, the CCC specification is irrelevant.
+func ShouldActuate(ctx context.Context, c client.Client, resourceNN types.NamespacedName) (bool, error) {
+	var cc opv1beta1.ConfigConnector
+	if err := c.Get(ctx, types.NamespacedName{
+		Name: opk8s.ConfigConnectorAllowedName,
+	}, &cc); err != nil {
+		if apierrors.IsNotFound(err) {
+			// if no CC exists, then by definition, KCC cannot be running in namespaced mode;
+			// so we can infer that we can use the default actuation action defined in code.
+			return ReconcilingByDefault(), nil
+		}
+		return false, fmt.Errorf("couldn't fetch ConfigConnector: %w", err)
+	}
+
+	if cc.Spec.Mode == opk8s.NamespacedMode {
+		var ccc opv1beta1.ConfigConnectorContext
+		if err := c.Get(ctx, types.NamespacedName{
+			Name:      opk8s.ConfigConnectorContextAllowedName,
+			Namespace: resourceNN.Namespace,
+		}, &ccc); err != nil {
+			if apierrors.IsNotFound(err) {
+				// if no CCC is defined in this namespace, defer to the "globally"
```

Comment:
It's probably going to be a lot more readable (and reusable) to first collect the `var cc *CC` and `var ccc *CCC`, then check them in the order of precedence, rather than mixing it into the fetching logic here